### PR TITLE
[JVM_IR] Use iinc for incrementing Int variables.

### PR DESCRIPTION
**Pull request by madsager**
Fix line number generation for assignments where the right-hand
side of the assignment is not on the same line.

Fix line number generation for intrinsics functions where the
function is not on the same line as the last argument.

Be careful to not break stepping behavior with the iinc
optimizations.

### CODE REVIEW

File: compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt [hunk 1]
```diff
@@ -591,9 +595,73 @@ class ExpressionCodegen(
...
+                expression.markLineNumber(startOffset = true)
+                mv.iinc(findLocalIndex(expression.symbol), if (isMinus) -delta else delta)
+                return true
+            }
+        }
+        return false
+    }
+
+    private fun hasSameLineNumber(element0: IrElement, element1: IrElement): Boolean
+        = getLineNumberForOffset(element0.startOffset) == getLineNumberForOffset(element1.startOffset)
```
Comment (dnpetrov) [inline compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt]: Minor, style: put `=` on the same line as function definition ( https://kotlinlang.org/docs/reference/coding-conventions.html#expression-body-formatting )

Comment (madsager) [reply compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt]: Thanks. Done.

**Comment by dnpetrov** [comment]
Ok, looks good to me.
I'll run tests and push manually.

**Comment by dnpetrov** [comment]
Pushed manually.

5570a5fe74b12dcd74f99447f5511534ca6963f3

**Pull request closed without merging**